digdag-standards/src/main/java/io/digdag/standards/operator/redshift/RedshiftConnection.java

```diff
@@ -0,0 +1,475 @@
+    }
+
+    String buildUnloadStatement(UnloadConfig unloadConfig)
+    {
+        StringBuilder sb = new StringBuilder();
+
+        sb.append(
+                String.format("UNLOAD ('%s') TO '%s'\n",
+                        escapeParam(unloadConfig.query),
+                        escapeParam(unloadConfig.toWithPrefixDir)));
+
+        // credentials
+        appendCredentialsPart(sb, unloadConfig);
+
+        // Options
+        appendOption(sb, "MANIFEST", unloadConfig.manifest);
+        appendOption(sb, "ENCRYPTED", unloadConfig.encrypted);
+        appendOption(sb, "DELIMITER", unloadConfig.delimiter);
+        appendOption(sb, "FIXEDWIDTH", unloadConfig.fixedwidth);
+        appendOption(sb, "GZIP", unloadConfig.gzip);
+        appendOption(sb, "BZIP2", unloadConfig.bzip2);
+        appendOption(sb, "ADDQUOTES", unloadConfig.addquotes);
+        appendOption(sb, "NULL AS", unloadConfig.nullAs);
+        appendOption(sb, "ESCAPE", unloadConfig.escape);
+        appendOption(sb, "ALLOWOVERWRITE", unloadConfig.allowoverwrite);
+        appendOption(sb, "PARALLEL", unloadConfig.parallel);
+
+        return sb.toString();
+    }
+
+    private class RedshiftPersistentTransactionHelper
+            extends PgPersistentTransactionHelper
+    {
+        RedshiftPersistentTransactionHelper(String statusTableName, Duration cleanupDuration)
+        {
+            super(statusTableName, cleanupDuration);
+        }
+
+        @Override
+        public void cleanup()
+        {
+            try (Statement stmt = connection.createStatement()) {
+                // List up status tables
+                List<String> statusTables = new ArrayList<>();
+                {
+                    ResultSet rs = stmt.executeQuery(
+                            String.format(ENGLISH,
+                                    "SELECT tablename FROM pg_tables WHERE tablename LIKE '%s_%%'",
+                                    escapeParam(statusTableName))
+                    );
+                    while (rs.next()) {
+                        statusTables.add(rs.getString(1));
+                    }
+                }
+
+                // Drop a status table if it is expired
+                statusTables.forEach(
+                        statusTable -> {
+                            try {
+                                ResultSet rs = stmt.executeQuery(
+                                        String.format(ENGLISH,
+                                                "SELECT query_id FROM %s WHERE created_at IS NOT NULL AND completed_at < CURRENT_TIMESTAMP - INTERVAL '%d SECOND'",
+                                                escapeIdent(statusTable),
+                                                cleanupDuration.getSeconds())
+                                );
+                                if (rs.next()) {
+                                    stmt.executeUpdate(String.format("DROP TABLE %s", escapeIdent(statusTable)));
+                                }
```

Comment:
Good catch 👍 

PostgreSQL's system catalog (e.g. pg_class) doesn't have table creation timestamp. So using `CREATE TABLE AS SELECT` is only an option to avoid creating an empty table. But I think we can go without it for the first release.